include/tudat/simulation/estimation_setup/processTrackingTxtFile.h

```diff
@@ -0,0 +1,319 @@
+/*    Copyright (c) 2010-2023, Delft University of Technology
+ *    All rights reserved
+ *
+ *    This file is part of the Tudat. Redistribution and use in source and
+ *    binary forms, with or without modification, are permitted exclusively
+ *    under the terms of the Modified BSD license. You should have received
+ *    a copy of the license with this file. If not, please or visit:
+ *    http://tudat.tudelft.nl/LICENSE.
+ *
+ *    References: 820-013, TRK-2-18 Tracking System Interfaces Orbit Data File Interface, Revision E, 2008, JPL/DSN
+ */
+
+
+#ifndef TUDAT_PROCESSTRACKINGTXTFILE_H
+#define TUDAT_PROCESSTRACKINGTXTFILE_H
+
+#include "tudat/basics/utilities.h"
+#include "tudat/io/readTrackingTxtFile.h"
+#include "tudat/astro/observation_models/observableTypes.h"
+#include "tudat/io/basicInputOutput.h"
+#include "tudat/astro/basic_astro/timeConversions.h"
+#include "tudat/astro/ground_stations/transmittingFrequencies.h"
+#include "tudat/simulation/estimation_setup/observations.h"
+#include "tudat/simulation/estimation_setup/observationSimulationSettings.h"
+#include "tudat/simulation/environment_setup/body.h"
+#include "tudat/math/interpolators/lookupScheme.h"
+
+// TODO:
+//  Ancillary settings (frequency, time scale, ...)
+//  Format checks
+//  Unit Checks
+//  Add time conversions
+
+
+
+namespace tudat
+{
+namespace observation_models
+{
+
+//! Utility function that might belong elsewhere
+template< typename TimeScalarType = double >
+TimeScalarType convertCalendarDateToJulianDaySinceJ2000(const int calendarYear,
+                                                        const int calendarMonth,
+                                                        const int calendarDay,
+                                                        const int calendarHour,
+                                                        const int calendarMinutes,
+                                                        const TimeScalarType calendarSeconds)
+{
+  return basic_astrodynamics::convertCalendarDateToJulianDaysSinceEpoch<TimeScalarType>
+      (calendarYear, calendarMonth, calendarDay, calendarHour, calendarMinutes,
+       calendarSeconds, basic_astrodynamics::JULIAN_DAY_ON_J2000);
+}
+
+//! Utility function to check if a container contains all elements of another container
```

Comment:
Could you move this function (and the next two) to the utilities.h file? That's where we usually collect these type of functions. They could also be useful for other things in the future :)

Reply:
Certainly.